### PR TITLE
JavaScript Builtins: Update Edge versioning information...

### PR DESCRIPTION
**Pull request by mdittmer**
... according to [Web API Confluence Dashboard](https://web-confluence.appspot.com/) dataset

### CODE REVIEW

**Comment by mdittmer** [comment]
I think I have a tentative explanation (though not a fix) for the `version_removed: 14` issue. [As mentioned here](https://github.com/GoogleChromeLabs/confluence/issues/202#issuecomment-332992863) Confluence skips prototype members that appear to be constants. However, I think the logic for detecting what constitutes a constant changed with a [major overhaul](https://github.com/GoogleChromeLabs/confluence/pull/220) of how APIs are gathered from JavaScript object graphs.

I suspect that the Edge 13 data are old enough that they contain less metadata that are used for detecting constants than subsequent versions; i.e., Edge 13+ includes the APIs in question, but are filtered out as "uninteresting constants" starting in Edge 14 (hence `version_removed: 14`).

Further investigation required to figure out how to sift this edge case out of the `mdittmer/mdn-confluence` output.

**Comment by erikadoyle** [comment]
- Looks like TypedArray.byteLength is incorrect (that's been around since IE10, so would definitely be in EdgeHTML 12)

**Comment by mdittmer** [comment]
In conversation with @patrickkettner it looks like timing properties of SharedArrayBuffers have lowered precision, but SharedArrayBuffer itself may still exist in latest Edge. We are still investigating the status of Atomics.

//CC @bterlson

**Comment by patrickkettner** [comment]
@Elchi3 @mdittmer did we need any more information to get this fixed and merged?

**Comment by Elchi3** [comment]
I'm still not convinced the "version_removed: 14" changes are correct. The original PR needs some updating still according the the comments above.

**Comment by mdittmer** [comment]
@patrickkettner is there anyone at Microsoft that could confirm or deny the individual `version_removed: 14` oddities? I can fix up based on individual comments, or if someone wants to create a derivative PR that can be merged, that works too.

File: javascript/builtins/String.json [hunk 1]
```diff
@@ -1130,7 +1130,8 @@
                 "version_added": true
               },
               "edge": {
-                "version_added": true
+                "version_added": true,
+                "version_removed": "14"
```
Comment (patrickkettner) [inline javascript/builtins/String.json]: we did not remove the length property from Strings

File: javascript/builtins/Error.json [hunk 2]
```diff
@@ -269,7 +269,8 @@
                 "version_added": true
               },
               "edge": {
-                "version_added": true
+                "version_added": true,
+                "version_removed": "14"
```
Comment (patrickkettner) [inline javascript/builtins/Error.json]:
```js
try {
    var e = new Error('test'); 
    throw e;
} catch (e) {
  console.log('message' in e)
}
```

gives true for edge 14 through current

File: javascript/builtins/Number.json [hunk 3]
```diff
@@ -322,7 +322,8 @@
                 "version_added": true
               },
               "edge": {
-                "version_added": true
+                "version_added": true,
+                "version_removed": "14"
```
Comment (patrickkettner) [inline javascript/builtins/Number.json]: We did not remove `NaN` from our browser

File: javascript/builtins/RegExp.json [hunk 4]
```diff
@@ -474,7 +474,8 @@
                 "version_added": true
               },
               "edge": {
-                "version_added": true
+                "version_added": true,
+                "version_removed": "14"
```
Comment (patrickkettner) [inline javascript/builtins/RegExp.json]: we did not remove `lastIndex`

**Comment by patrickkettner** [comment]
@mdittmer done!

**Comment by mdittmer** [comment]
Dropped bad `version_removed:14`s. @Elchi3 PTAL.

File: javascript/builtins/String.json (hunk 1 above)
Comment (mdittmer) [reply javascript/builtins/String.json]: This (and possible other) error(s) appears to come from our [exclude constants from API list heuristic](https://github.com/GoogleChromeLabs/confluence/blob/e10a30d8e346c0836952d7632bc27d7f7756c36c/lib/web_catalog/api_extractor.es6.js#L680). `Object.getOwnPropertyDescriptor(String.prototype, 'length').value` in Edge returns `0`, not `undefined`.

Comment (mdittmer) [reply javascript/builtins/String.json]: @foolip and I are considering combining the existing heuristic with a `CONST_CASE` name check to get a better guess at "constants that can be ignored in method+property counts".

**Comment by erikadoyle** [comment]
Thank you so much @mdittmer!!

**Pull request merged**